ush/ioda/bufr2ioda/bufr2ioda_subpfl_argo_profiles.py

```diff
@@ -0,0 +1,323 @@
+from wxflow import Logger
+
+
+def Compute_sequenceNumber(lon):
+    lon_u, seqNum = np.unique(lon, return_inverse=True)
+    seqNum = seqNum.astype(np.int64)
+    logger.debug(f"Len of Sequence Number: {len(seqNum)}")
+
+    return seqNum
+
+
+def bufr_to_ioda(config, logger):
+
+    subsets = config["subsets"]
+    logger.debug(f"Checking subsets = {subsets}")
+
+    # Get parameters from configuration
+    data_format = config["data_format"]
+    source = config["source"]
+    data_type = config["data_type"]
+    data_description = config["data_description"]
+    data_provider = config["data_provider"]
+    cycle_type = config["cycle_type"]
+    cycle_datetime = config["cycle_datetime"]
+    dump_dir = config["dump_directory"]
+    ioda_dir = config["ioda_directory"]
+    cycle = config["cycle_datetime"]
+
+    yyyymmdd = cycle[0:8]
+    hh = cycle[8:10]
+
+    # General Information
+    converter = 'BUFR to IODA Converter'
+    platform_description = 'ARGO profiles from subpfl: temperature and salinity'
+    bufrfile = f"{cycle_type}.t{hh}z.{data_format}.tm{hh}.bufr_d"
+    DATA_PATH = os.path.join(dump_dir, f"{cycle_type}.{yyyymmdd}", str(hh), f"atmos",
+                             bufrfile)
+
+    # ===========================================
+    # Make the QuerySet for all the data we want
+    # ===========================================
+    start_time = time.time()
+
+    logger.debug(f"Making QuerySet ...")
+    q = bufr.QuerySet()
+
+    # MetaData
+    q.add('year', '*/YEAR')
+    q.add('month', '*/MNTH')
+    q.add('day', '*/DAYS')
+    q.add('hour', '*/HOUR')
+    q.add('minute', '*/MINU')
+    q.add('ryear', '*/RCYR')
+    q.add('rmonth', '*/RCMO')
+    q.add('rday', '*/RCDY')
+    q.add('rhour', '*/RCHR')
+    q.add('rminute', '*/RCMI')
+    q.add('stationID', '*/WMOP')
+    q.add('latitude', '*/CLATH')
+    q.add('longitude', '*/CLONH')
+    q.add('pressure', '*/GLPFDATA/WPRES')
+
+    # ObsValue
+    q.add('temp', '*/GLPFDATA/SSTH')
+    q.add('saln', '*/GLPFDATA/SALNH')
+
+    end_time = time.time()
+    running_time = end_time - start_time
+    logger.debug(f"Running time for making QuerySet: {running_time} seconds")
+
+    # ===============================================================
+    # Open the BUFR file and execute the QuerySet
+    # Use the ResultSet returned to get numpy arrays of the data
+    # ==============================================================
+
+    logger.debug(f"Executing QuerySet to get ResultSet ...")
+    with bufr.File(DATA_PATH) as f:
+        r = f.execute(q)
+
+    # MetaData
+    logger.debug(f" ... Executing QuerySet: get MetaData ...")
+    dateTime = r.get_datetime('year', 'month', 'day', 'hour', 'minute', group_by='pressure')
+    dateTime = dateTime.astype(np.int64)
+    rcptdateTime = r.get_datetime('ryear', 'rmonth', 'rday', 'rhour', 'rminute', group_by='pressure')
+    rcptdateTime = rcptdateTime.astype(np.int64)
+    stationID = r.get('stationID', group_by='pressure')
```

Comment:
Just a quick note, the "group_by" is not required, but this is still fine.

Reply:
Thanks @nicholasesposito